@xen-orchestra/xapi/vdi.mjs

```diff
@@ -134,22 +134,22 @@ class Vdi {
     if (stream.length === undefined) {
       throw new Error('Trying to import a VDI without a length field. Please report this error to Xen Orchestra.')
     }
+    const vdi = await this.getRecord('VDI', ref)
+    const sr = await this.getRecord('SR', vdi.SR)
```

Comment:
the shouldn't but they can throw an error if object is not already in memory , you should move this in the try catch or handle this case separatly